actionview/lib/action_view/helpers/tags/collection_select.rb

```diff
@@ -14,16 +14,24 @@ def initialize(object_name, method_name, template_object, collection, value_meth
         end
 
         def render
-          option_tags_options = {
-            selected: @options.fetch(:selected) { value },
-            disabled: @options[:disabled]
-          }
-
-          select_content_tag(
-            options_from_collection_for_select(@collection, @value_method, @text_method, option_tags_options),
-            @options, @html_options
-          )
+          select_content_tag(option_tags, @options, @html_options)
         end
+
+        private
+          def collection_grouped?
+            entry = @collection.first if @collection.is_a?(Hash) || @collection.is_a?(Array)
+            entry.respond_to?(:last) && entry.last.is_a?(Array)
+          end
```

Comment:
Sure!  The reason I limited this behavior to `Hash` and `Array` was because I did not realize I could use `blank?` to ensure that an AR relation was loaded.  I had assumed `blank?` would execute a limited query, just like `empty?`.  It looks like this behavior was explicitly implemented in #5461 (and tests were added in #5469), so I will change this method to mirror `Select`. :+1: